pkg/admission/util.go

```diff
@@ -37,26 +37,40 @@ func updatePodLabel(pod *v1.Pod, namespace string, generateUniqueAppIds bool, de
 	}
 
 	sparkAppID := utils.GetPodLabelValue(pod, constants.SparkLabelAppID)
+	canonicalAppID := utils.GetPodLabelValue(pod, constants.CanonicalLabelApplicationID)
 	appID := utils.GetPodLabelValue(pod, constants.LabelApplicationID)
-	if sparkAppID == "" && appID == "" {
+	if canonicalAppID == "" && sparkAppID == "" && appID == "" {
 		// if app id not exist, generate one
 		// for each namespace, we group unnamed pods to one single app - if GenerateUniqueAppId is not set
 		// if GenerateUniqueAppId:
 		//		application ID convention: ${NAMESPACE}-${POD_UID}
 		// else
 		// 		application ID convention: ${AUTO_GEN_PREFIX}-${NAMESPACE}-${AUTO_GEN_SUFFIX}
 		generatedID := utils.GenerateApplicationID(namespace, generateUniqueAppIds, string(pod.UID))
+
+		result[constants.CanonicalLabelApplicationID] = generatedID
```

Comment:
Do we need to apply `constants.CanonicalLabelApplicationID` to `GetApplicationIDFromPod`? It seems `GetApplicationIDFromPod` don't take app id from this new label

Reply:
Hi @chia7712, thanks for the review.  
I'm going to change GetApplicationIDFromPod() and GetQueueNameFromPod() in [YUNIKORN-2504 ](https://issues.apache.org/jira/browse/YUNIKORN-2504).  This PR is focus on Admission Controller change. These two function are not called in admission controller.

Reply:
Base on your feedback. I think there have a bug in current Admission Controller and I believe it should be fixed in another PR.
-[ [YUNIKORN-2636] Admission Controller ignores existing Queue/ApplicationID annotations ](https://issues.apache.org/jira/browse/YUNIKORN-2636)

Reply:
Merged with the changes in YUNIKORN-2636. 

Based on the same reason in YUNIKORN-2636's discussion.
- https://github.com/apache/yunikorn-k8shim/pull/848#discussion_r1611634390

I'm not going to chang to use GetApplicationIDFromPod()/GetQueueNameFromPod() in Admission Controller.